src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T,V> {
 
+    private T first;
+    private V second;
+
+    private Pair(T first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public V getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<T, V> pair = (Pair) o;
+        return (first == null || first.equals(pair.first))
```

Comment:
what would be the result if first and second == null and pairt.first and pair.second != null ?